### ISSUE TITLE
Add Log-Softmax Temperature Option to transformers.Seq2SeqTrainingArguments for CE Loss

### ISSUE THREAD

**Issue by oswaldoludwig** [huggingface/transformers#33267]
### Feature request

I would like to propose an enhancement to the transformers.Seq2SeqTrainingArguments class. Specifically, I suggest including an option to adjust the log-softmax temperature for the cross-entropy (CE) loss function. 

### Motivation

This feature would be particularly beneficial for training models with large vocabularies, as highlighted [here](https://medium.com/@oswaldoludwig/unveiling-a-pitfall-in-cross-entropy-loss-for-large-vocabularies-ef388d846bc1)

### Your contribution

For Whisper fine-tuning, I worked around this issue by using the following class:

```
class CustomSeq2SeqTrainer(Seq2SeqTrainer):
    def __init__(self, *args, temp=1.0, **kwargs):
        super().__init__(*args, **kwargs)
        self.temp = temp

    def compute_loss(self, model, inputs, return_outputs=False):
        labels = inputs.get("labels")
        # Forward pass
        outputs = model(**inputs)
        logits = outputs.get("logits")

        # Apply temperature scaling
        logits = logits / self.temp

        # Compute loss
        loss_fct = torch.nn.CrossEntropyLoss()
        loss = loss_fct(logits.view(-1, logits.size(-1)), labels.view(-1))

        return (loss, outputs) if return_outputs else loss

```

**Comment by LysandreJik** [comment]
Hey @oswaldoludwig, the trainer implementation is made to be extensible as it cannot ever satisfy every request; the way you did it is great!

If we see significant interest in this issue we can see about upstreaming it; if a reader is interested in having this in transformers as well, please comment here and we'll consider it.
